### ISSUE TITLE
Type error when exporting array of columns generated by the column helper

### ISSUE THREAD

**Issue by C-Higgins** [TanStack/table#5135]
### Describe the bug

When exporting an array of columns under certain conditions, a type error appears in the IDE (not in compiler). This happens across multiple IDEs, computers, and people. It only happens when the array is exported and there are two or more different string-extending types, such as a string and a string|null, or a string and a string literal.

Type error: 
```
TS4023: Exported variable  cols  has or is using name  AccessorKeyColumnDefBase  from external module
"node_modules/@tanstack/table-core/build/lib/types"
but cannot be named.
TS4023: Exported variable  cols  has or is using name  IdIdentifier  from external module
"node_modules/@tanstack/table-core/build/lib/types"
but cannot be named.
TS4023: Exported variable  cols  has or is using name  StringHeaderIdentifier  from external module
"node_modules/@tanstack/table-core/build/lib/types"
but cannot be named.
```

### Your minimal, reproducible example

This cannot be reproduced in Stackblitz or the like. It only appears in Intellisense, not the TS compiler

### Steps to reproduce

If the data type has two or more stringlike types that differ, and the column array is exported, an error appears. Any of the following will produce the error:

```typescript
const cd = createColumnHelper<{a: 'a'; b:string}>();
export const cols = [
  cd.accessor("a", {}),
  cd.accessor("b", {}),
];
```

```typescript
const cd = createColumnHelper<{a: string|null; b:string}>();
export const cols = [
  cd.accessor("a", {}),
  cd.accessor("b", {}),
];
```

etc

### Expected behavior

No error

### How often does this bug happen?

Every time

### Platform

Jetbrains IDE and VSCode IDE

### react-table version

8.10.5

### TypeScript version

5.2.2

### Additional context

Related PR: #5133 

### Terms & Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I understand that if my bug cannot be reliable reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by sebastiangug** [comment]
@C-Higgins  did you ever find any fix for this? going through the same right now.

**Comment by C-Higgins** [comment]
> @C-Higgins did you ever find any fix for this? going through the same right now.

The fix is in #5133, though the maintainers don't seem to be interested in merging it
